applications/tari_base_node/src/grpc/server.rs

```diff
@@ -666,7 +776,272 @@ impl From<BlockHeader> for base_node_grpc::BlockHeader {
                 accumulated_monero_difficulty: h.pow.accumulated_monero_difficulty.into(),
                 accumulated_blake_difficulty: h.pow.accumulated_blake_difficulty.into(),
                 pow_data: h.pow.pow_data,
+                target_difficulty: h.pow.target_difficulty.as_u64(),
+            }),
+        }
+    }
+}
+
+impl From<ChainMetadata> for base_node_grpc::MetaData {
+    fn from(meta: ChainMetadata) -> Self {
+        let diff = meta.accumulated_difficulty.unwrap_or_else(Difficulty::min);
+        Self {
+            height_of_longest_chain: meta.height_of_longest_chain.unwrap_or(0),
+            best_block: meta.best_block.unwrap_or(vec![]),
+            pruning_horizon: meta.pruning_horizon,
+            accumulated_difficulty: diff.as_u64(),
+        }
+    }
+}
+
+impl From<NewBlockTemplate> for base_node_grpc::NewBlockTemplate {
```

Comment:
I think put these conversions in a new file, or even separate files. This file is getting pretty big and these may be used elsewhere